src/Lecture2.hs

```diff
@@ -48,7 +49,10 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct [] = 1
+lazyProduct (x:xs)
+  | x == 0 = 0
+  | otherwise = x * lazyProduct xs
```

Comment:
Correct solution 👍🏻 Alternatively, you can match on zero in the pattern as well

```suggestion
lazyProduct (0 : _) = 0
```

Reply:
More elegant, thanks!